src/main/java/swinglib/JPanelBuilder.java

```diff
@@ -0,0 +1,191 @@
+package swinglib;
+
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.GridBagLayout;
+import java.awt.GridLayout;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import javax.swing.BoxLayout;
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+import javax.swing.border.EmptyBorder;
+
+import org.apache.commons.math3.util.Pair;
+
+/**
+ * Example usage:
+ * <code><pre>
+ *   final JPanel panel = JPanelBuilder.builder()
+ *       .gridLayout(2, 1)
+ *       .add(new JLabel("")
+ *       .add(new JLabel("")
+ *       .build();
+ * </pre></code>
+ */
+public class JPanelBuilder {
+
+  private final Collection<Pair<Component, BorderLayoutPosition>> components = new ArrayList<>();
+  private Layout layout = Layout.DEFAULT;
+  private int gridRows;
+  private int gridColumns;
+  private BorderType borderType;
+  private int borderWidth;
+  private Float xAlignment;
```

Comment:
Hmm, I think we may want to update that rule, `xAlignment` does seem okay, but I did note the checkstyle violation.